src/interpreter/type_cast.rs

```diff
@@ -0,0 +1,224 @@
+use crate::interpreter::types::{Numeric, PactType, StringLike};
+use core::any::Any;
+use core::convert::TryFrom;
+use std::string::String;
+use std::vec::Vec;
+
+#[derive(Debug, PartialEq)]
+pub enum PactConversionErr {
+    Overflow,
+    UnknownType,
+}
+
+/// A catch-all conversion trait which tries to turn any given `value` into the implementing type
+pub trait AnyTryInto<'a>: Sized {
+    fn any_try_into(value: &'a dyn Any) -> Result<Self, PactConversionErr>;
+}
+
+/// AnyTryInto implementation for PactType
+impl<'a> AnyTryInto<'a> for PactType<'a> {
+    fn any_try_into(value: &'a dyn Any) -> Result<PactType<'a>, PactConversionErr> {
+        // TODO: refactor the below repetiion using macros
+        // Unsigned integer type casting into PactType
+        if let Some(number) = value.downcast_ref::<u8>() {
+            if let Ok(n) = u64::try_from(*number) {
```

Comment:
It is safe and more efficient to use an `as` cast for `u8`, `u16`, and `u32` as they will never exceed `u64`